File: 刷题/golang/55. 跳跃游戏.go
func canJump(nums []int) bool {
    maxD:=0
    n:=len(nums)
    for i := range nums {
        if i<=maxD {
            maxD = max(maxD, i+nums[i])
            if maxD >= n-1 {
                return true
            }
        }
    }
    return false
}
func max(x, y int) int {
    if x>y {
        return x
    }
    return y
}

File: 刷题/cpp/55. 跳跃游戏.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxD=0;
        for(int i=0;i<nums.size();i++){
            if(i<=maxD){
                maxD=max(maxD,i+nums[i]);
                if(maxD>=nums.size()-1) return true;
            }
        }
        return false;
    }
};